assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -15519,4 +15531,54 @@ public void setSubmissionTimestamp(String timeStamp) {
             m_timeStamp = timeStamp;
         }
     }
+
+    /**
+     * Used when students are uploading a video object
+     */
+    private List<Reference> getVideoAttachments(RunData data, SessionState state, Assignment assignment) {
+        List<Reference> attachments = entityManager.newReferenceList();
+        //Get the video object in base64 from the form post
+        String videoResponseBase64 = data.getParameters().get("videoResponse");
+        if(StringUtils.isBlank(videoResponseBase64)){
+            return attachments;
+        }
+        videoResponseBase64 = videoResponseBase64.split(",")[1];
+        // Use the sessionId for the video reference.
+        String resourceId = sessionManager.getCurrentSession().getId();
+        String contentType = "video/webm";
+        // The blob is encoded Base64 in the client side and sent via POST.
+        InputStream contentStream = new ByteArrayInputStream(Base64.getDecoder().decode(videoResponseBase64.getBytes(StandardCharsets.UTF_8)));
+        ResourcePropertiesEdit props = contentHostingService.newResourceProperties();
+        props.addProperty(ResourceProperties.PROP_DISPLAY_NAME, resourceId);
+        props.addProperty(ResourceProperties.PROP_DESCRIPTION, resourceId);
+        SecurityAdvisor sa = createSubmissionSecurityAdvisor();
+        try {
+            securityService.pushAdvisor(sa);
+            String siteId = toolManager.getCurrentPlacement().getContext();
+            ContentResource attachment = contentHostingService.addAttachmentResource(resourceId, siteId, "Assignments", contentType, contentStream, props);
+            try {
+                Reference ref = entityManager.newReference(contentHostingService.getReference(attachment.getId()));
+                attachments.add(ref);
+                state.setAttribute(ATTACHMENTS, attachments);
+            } catch (Exception e) {
+                log.warn("Cannot find reference for {}: {}", attachment.getId(), e.getMessage());
+            }
+        } catch (PermissionException e) {
+            addAlert(state, rb.getString("notpermis4"));
+        } catch (RuntimeException e) {
+            if (contentHostingService.ID_LENGTH_EXCEPTION.equals(e.getMessage())) {
+                addAlert(state, rb.getFormattedMessage("alert.toolong", resourceId));
+            }
```

Comment:
You should probably rethrow this if it's not an ID_LENGTH_EXCEPTION. In fact, BaseContentService should not be throwing a runtime exception. It should, imo, throw an IllegalArgumentException. Anyway, it's a bad idea to consume runtime exceptions.

Reply:
Correct, I'm just replicating what's it in the source.